src/blockchain/xalpha/burn.ts

```diff
@@ -0,0 +1,46 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { ContractTransaction } from '@ethersproject/contracts'
+import { BaseProvider } from '@ethersproject/providers'
+import { ADDRESSES, ALPHA, ETH } from '@xtoken/abis'
+import { ethers } from 'ethers'
+
+import {
+  DEC_18,
+  GAS_LIMIT_PERCENTAGE_DEFAULT,
+  GAS_LIMIT_PERCENTAGE_ETH,
+} from '../../constants'
+import { XALPHA } from '../../types'
+import { ITokenSymbols } from '../../types/xToken'
+import { getPercentage } from '../../utils'
+import { getExpectedRate, parseFees } from '../utils'
+
+import { getXAlphaContracts } from './helper'
+import { getXAlphaPrices } from './prices'
+
+const { formatEther, parseEther } = ethers.utils
+
+// TODO: should be using Uniswap for expected quantities?
```

Comment:
Nope, that'd be a different estimate under `exchanges` module for available liquidity pool. You're missing `getExpectedQuantityOnBurnXAlpha()` method implementation here

Reply:
Thanks, added this method.